source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2736 @@
+  in order to perform these assertions.
+
+  Test files SHOULD NOT expect events from multiple specs (e.g. command
+  monitoring *and* SDAM events) for a single client. See
+  `Mixing event types in observeEvents and expectEvents`_ for more
+  information.
+
+.. _test_outcome:
+
+- ``outcome``: Optional array of one or more `collectionData`_ objects. Data
+  that should exist in collections after each test case is executed.
+
+  The list of documents herein SHOULD be sorted ascendingly by the ``_id`` field
+  to allow for deterministic comparisons. The procedure for asserting collection
+  contents is discussed in `Executing a Test`_.
+
+
+operation
+~~~~~~~~~
+
+An operation to be executed as part of the test.
+
+The structure of this object is as follows:
+
+.. _operation_name:
+
+- ``name``: Required string. Name of the operation (e.g. method) to perform on
+  the object.
+
+.. _operation_object:
+
+- ``object``: Required string. Name of the object on which to perform the
+  operation. This should correspond to either an `entity`_ name (for
+  `Entity Test Operations`_) or "testRunner" (for `Special Test Operations`_).
+  If the object is an entity, The YAML file SHOULD use an `alias node`_ for its
+  ``id`` field (e.g. ``object: *collection0``).
+
+.. _operation_arguments:
+
+- ``arguments``: Optional object. Map of parameter names and values for the
+  operation. The structure of this object will vary based on the operation.
+  See `Entity Test Operations`_ and `Special Test Operations`_.
+
+  The ``session`` parameter is handled specially (see `commonOptions_session`_).
+
+.. _operation_expectError:
+
+- ``expectError``: Optional `expectedError`_ object. One or more assertions for
+  an error expected to be raised by the operation.
+
+  This field is mutually exclusive with
+  `expectResult <operation_expectResult_>`_ and
+  `saveResultAsEntity <operation_saveResultAsEntity_>`_.
+
+.. _operation_expectResult:
+
+- ``expectResult``: Optional mixed type. A value corresponding to the expected
+  result of the operation. This field may be a scalar value, a single document,
+  or an array of documents in the case of a multi-document read.  Test runners
+  MUST follow the rules in `Evaluating Matches`_ when processing this assertion.
+
+  This field is mutually exclusive with `expectError <operation_expectError_>`_.
+
+.. _operation_saveResultAsEntity:
+
+- ``saveResultAsEntity``: Optional string. If specified, the actual result
+  returned by the operation (if any) will be saved with this name in the
+  `Entity Map`_.  The test runner MUST raise an error if the name is already in
+  use or if the result is not a `Supported Entity Type`_.
+
+  This field is mutually exclusive with `expectError <operation_expectError_>`_.
+
+
+expectedError
+~~~~~~~~~~~~~
+
+One or more assertions for an error/exception, which is expected to be raised by
+an executed operation. At least one key is required in this object.
```

Comment:
Is any combination of the fields allowed? Is the test runner meant to sequentially check each of the listed fields and take the actions listed therein? If not, some implementation guidance here would be helpful.

Reply:
Multiple fields may be specified. Each field is a different assertion and `expectError` requires at least one be specified (`isError` being the most basic).

Reply:
SGTM.